src/lib/Hydra/Controller/Jobset.pm

```diff
@@ -261,14 +261,22 @@ sub updateJobset {
 
     my $checkinterval = int(trim($c->stash->{params}->{checkinterval}));
 
+    my $enable_dynamic_run_command = defined $c->stash->{params}->{enable_dynamic_run_command} ? 1 : 0;
+    if ($enable_dynamic_run_command
+        && !($c->config->{dynamicruncommand}->{enable}
+            && $jobset->project->enable_dynamic_run_command))
+    {
+        badRequest($c, "Dynamic RunCommand is not enabled by the server or the parent project.");
+    }
+
```

Comment:
> I'm not sure what the right thing here is, but this pattern of checking has been peppered around. Instead we should probably have a single function which implements "is dynamic run command enabled?" and use that everywhere.

https://github.com/DeterminateSystems/hydra/pull/2#discussion_r781260919

Reply:
For this, I had been experimenting with passing in the project ref itself and getting the `enable_dynamic_run_command` value from there (if the project was defined). However, the two places where a project is used (`AddBuilds.pm` and `Jobsets.pm`) had two different "definitions" of a project -- in `AddBuilds`, it was just a plain Perl hash, but in `Projects`, it was a `DBIx` thing.

This is an issue because you can only access `enable_dynamic_run_command` on a hash with `$project->{enable_dynamic_run_command}`, but for a `DBIx` thing, you must use `$project->enable_dynamic_run_command` (note the lack of curly braces).

One workaround I attempted was to get a hash from the `$project` `DBIx` object via `{$project->get_columns}`, but that looks pretty ugly and would be difficult to intuit if this were to ever happen elsewhere.

So then the only real way to factor this function out into a common location would be to make it take 2 arguments: whether the request wants it to be enabled (e.g. a declarative jobset or a API POST), and if it's enabled on the parent project. For projects themselves, this latter argument would be `undef` (and we'd check for that), and then we'd also check if it's enabled in the Hydra config before returning `0` if it was disabled somewhere with higher precedence, or `1` if everything checks out.

I think this is the best way to go about it at present, but am open to other ideas.